src/table.tsx

```diff
@@ -75,83 +127,207 @@ export const Table = React.forwardRef(
+                                                idx === heading.length - 1 && !hasActionsColumn,
+                                        }
+                                    )}
+                                    style={{minWidth: columnWidth?.[idx], width: columnWidth?.[idx]}}
+                                >
+                                    {header}
+                                </th>
+                            ))}
+                            {hasActionsColumn && (
+                                <th>
+                                    <ScreenReaderOnly>
+                                        {/** TODO: use translated text for actions */}
+                                        <div className={styles.actionsHeaderText}>Actions</div>
+                                    </ScreenReaderOnly>
+                                </th>
+                            )}
+                        </Text1>
+                    </thead>
+                )}
                 <tbody>
-                    {content && content.length ? (
-                        content.map((row, rowIdx) => (
-                            <tr key={rowIdx}>
-                                {row.map((cell, idx) => (
-                                    <td key={idx} className={styles.verticalAlign[rowVerticalAlign]}>
-                                        {/**
-                                         * In collapsedRowsMode, we render the row heading text before every cell content, except for the first cell
-                                         * of every row, which is rendered with a medium weight font, as it's the row title.
-                                         * */}
-                                        {idx !== 0 && collapsedRowsMode && (
-                                            // this is aria-hidden because screen readers already read the column heading from the th
-                                            <div className={styles.mobileCellHeading} aria-hidden>
-                                                <Text1 medium color={vars.colors.textSecondary}>
-                                                    {heading[idx]}
-                                                </Text1>
-                                            </div>
-                                        )}
-
-                                        <Text2
-                                            as="div"
-                                            weight={idx === 0 && collapsedRowsMode ? 'medium' : 'regular'}
-                                            wordBreak={false}
+                    {content.length > 0 ? (
+                        content.map((row, rowIdx) => {
+                            const rowCells = !Array.isArray(row) ? row.cells : row;
+                            const rowActionsList = !Array.isArray(row) ? row.actions ?? [] : [];
+
+                            const actions = (
+                                <Inline space={16}>
```

Comment:
maybe don't create this Inline if there isn't any action?

Reply:
This element is rendered only when `rowActionsList.length > 0`, so the Inline doesn't exist in that case